src/SpendPermissionManager.sol

```diff
@@ -259,22 +272,28 @@ contract SpendPermissionManager is EIP712 {
         }
 
         // loop through each spend permission in the batch and approve it
+        bool allApproved = true;
         uint256 batchLen = spendPermissionBatch.permissions.length;
         for (uint256 i; i < batchLen; i++) {
-            _approve(
-                SpendPermission({
-                    account: spendPermissionBatch.account,
-                    spender: spendPermissionBatch.permissions[i].spender,
-                    token: spendPermissionBatch.permissions[i].token,
-                    allowance: spendPermissionBatch.permissions[i].allowance,
-                    period: spendPermissionBatch.period,
-                    start: spendPermissionBatch.start,
-                    end: spendPermissionBatch.end,
-                    salt: spendPermissionBatch.permissions[i].salt,
-                    extraData: spendPermissionBatch.permissions[i].extraData
-                })
-            );
+            if (
+                !_approve(
+                    SpendPermission({
```

Comment:
This is fairly nested. Maybe add a comment above if to make it a bit clearer what we're trying to do?